libs/langchain/langchain/schema/runnable.py

```diff
@@ -759,6 +759,109 @@ async def astream(
             yield item
 
 
+class RouterInput(TypedDict):
+    key: str
+    input: Any
+
+
+class RouterRunnable(Runnable[RouterInput, Output]):
```

Comment:
'router' sounds like the thing choosing the routing to me. this is more of a 'route executor'